src/test/java/org/elasticsearch/index/merge/policy/MergePolicySettingsTest.java

```diff
@@ -156,6 +159,50 @@ public void testUpdateSettings() throws IOException {
         }
     }
 
+    public void testTieredMergePolicySettingsUpdate() throws IOException {
+        IndexSettingsService service = new IndexSettingsService(new Index("test"), EMPTY_SETTINGS);
+        TieredMergePolicyProvider mp = new TieredMergePolicyProvider(createStore(EMPTY_SETTINGS), service);
+        assertThat(mp.getMergePolicy().getNoCFSRatio(), equalTo(0.1));
+
+        assertEquals(mp.getMergePolicy().getForceMergeDeletesPctAllowed(), TieredMergePolicyProvider.DEFAULT_EXPUNGE_DELETES_ALLOWED, 0.0d);
```

Comment:
I think we need to add a check that other settings do not spring back to their defaults, which I believe was the  original issue (update one setting and have all the rest go back to default).